src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,204 @@
+            }
+            for (Order order : orderList) {
+                order.setProducts(extractProductForOrder(order.getId(), connection));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Can't get all orders for user with id = " + userId, e);
+        }
+        return orderList;
+    }
+
+    @Override
+    public Order create(Order order) {
+        String query = "INSERT INTO orders (user_id)"
+                        + " VALUES (?)";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setLong(1, order.getUserId());
+            preparedStatement.executeUpdate();
+            ResultSet resultKey = preparedStatement.getGeneratedKeys();
+            if (resultKey.next()) {
+                order.setId(resultKey.getLong(1));
+            }
+            insertProducts(order, connection);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Order for user with id " + order.getUserId()
+                    + " has not been added", e);
+        }
+        return order;
+    }
+
+    private int insertProducts(Order order, Connection connection) throws SQLException {
+        String query = "INSERT INTO orders_products (orders_id, products_id) "
+                         + "VALUES (?, ?)";
+
+        int result = 0;
+        for (Product product : order.getProducts()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, order.getId());
+            preparedStatement.setLong(2, product.getId());
+            result += preparedStatement.executeUpdate();
+        }
+        return result;
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders"
+                        + " JOIN orders_products ON orders.order_id"
+                        + " = orders_products.orders_id"
+                        + " JOIN products ON products.product_id"
+                        + " = orders_products.products_id"
+                        + " where orders.order_id = ? ";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            if (sqlResult.next()) {
+                Order order = parseResult(sqlResult);
+                return Optional.of(order);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get order by id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    private Order parseResult(ResultSet sqlResult) throws SQLException {
```

Comment:
looks like you are parsing from ResultSet - an order. Should you name this method as parseOrder?

Reply:
Ok. Done